tables/DataMan/DataManInfo.cc

```diff
@@ -383,11 +382,24 @@ void DataManInfo::makeUniqueNames (Record& dminfo)
           name = cols[0];
         }
       }
+      dm.define ("NAME", name);
     }
-    // Make the name unique if needed.
-    String newName = uniqueName (dminfo, name, i);
-    if (newName != origName) {
-      dm.define ("NAME", newName);
+  }
+  // Now make the names unique if needed.
+  // The first instance is kept as is, others get suffix _1, _2, etc.
+  std::set<String> firstNames;
+  for (uInt i=0; i<dminfo.nfields(); ++i) {
+    Record& dm = dminfo.rwSubRecord(i);
+    String name = dm.asString("NAME");
+    if (firstNames.find(name) == firstNames.end()) {
+      // The first instance of this name is kept as is.
+      firstNames.insert (name);
+    } else {
+      // Make the name unique by adding the suffix.
+      String newName = uniqueName (dminfo, name, i);
+      if (newName != name) {
```

Comment:
`(newName != name)` should always be true here, shouldn't it? I therefore removed it in my suggestion in my other comment.

Reply:
Correct; changed